Added support for scala 2.12.

argast opened the pull request:
Added support for scala 2.12.

- Added crossbuild
- Upgraded libraries
- Added travis build
- Added automatic publishing from master
- Updated documentation

argast committed:
Added support for scala 2.12.

Added support for scala 2.12.

- Added crossbuild
- Upgraded libraries
- Added travis build
- Added automatic publishing from master
- Updated documentation

argast commented:
Hey Guys,

I updated library to include scala 2.12 support. Before you merge, you need to enable travis build and add following properties in order to make publishing work:

BINTRAY_USER
BINTRAY_API_KEY

There is a script run as part of build that will create a ~/.bintray/.credentials file.

Let me know if you have any questions.

kcaipq commented on .travis.yml:
```diff
@@ -0,0 +1,29 @@
+  sbt ++$TRAVIS_SCALA_VERSION test
+
+after_success:
+  - if [[ ( "$TRAVIS_BRANCH" == "master" || "$TRAVIS_PULL_REQUEST" == "false" ) ]]; then
```
I really don't think that having it published every time when a commit is made into master branch is a good idea. I suggest to use "TRAVIS_TAG" to restrict the publish only applies to tagged commits.

argast replied:
Good point. In that case it would make sense to switch to similar versioning logic we use other projects, using git tag. I will make that change.

enekomunoz commented:
@afiore @ostapneko I still have access to this repo. You might want to take me out

argast committed:
Finished travis setup

- Removed logging of bintray credentials
- Added build status
- Switched to timeoutdigital org
- Added information how to add dependency to sbt build.

argast committed:
Finished travis setup

- Removed logging of bintray credentials
- Added build status
- Switched to timeoutdigital org
- Added information how to add dependency to sbt build.

ostapneko commented:
@enekomunoz It's a public repo :)

argast committed:
Changed travis to publish only on tagged commits.

argast commented:
Now it should only publish from tags. E.g. v1.2.3
